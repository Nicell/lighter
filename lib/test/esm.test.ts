import { highlight } from "../dist/index.esm.mjs";
import { runTests } from "./highlight.js";

runTests({ highlight });
