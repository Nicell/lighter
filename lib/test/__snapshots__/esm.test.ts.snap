// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`highlight js 1`] = `
{
  "colors": {
    "activeTabBackground": "#1E1E1E",
    "activeTabBorder": "#1E1E1E",
    "activeTabForeground": "#ffffff",
    "activeTabTopBorder": undefined,
    "background": "#1E1E1E",
    "colorScheme": "dark",
    "diffInsertedLineBackground": "#9bb95533",
    "diffInsertedTextBackground": "#9ccc2c33",
    "diffRemovedLineBackground": "#ff000033",
    "diffRemovedTextBackground": "#ff000033",
    "editorBackground": "#1E1E1E",
    "editorGroupHeaderBackground": "#252526",
    "foreground": "#D4D4D4",
    "hoverTabBackground": "#2D2D2D",
    "hoverTabForeground": "#ffffff80",
    "iconForeground": "#C5C5C5",
    "inactiveTabBackground": "#2D2D2D",
    "inactiveTabForeground": "#ffffff80",
    "lineNumberForeground": "#858585",
    "listHoverBackground": "#2A2D2E",
    "listHoverForeground": undefined,
    "listSelectionBackground": "#37373D",
    "listSelectionForeground": undefined,
    "selectionBackground": "#264F78",
    "sideBarBackground": "#252526",
    "sideBarBorder": "#252526",
    "sideBarForeground": "#D4D4D4",
    "tabBorder": "#252526",
    "tabsBorder": undefined,
  },
  "lang": "javascript",
  "lines": [
    [
      {
        "content": "x",
        "style": {
          "color": "#9CDCFE",
        },
      },
      {
        "content": " = ",
        "style": {
          "color": "#D4D4D4",
        },
      },
      {
        "content": "1",
        "style": {
          "color": "#B5CEA8",
        },
      },
    ],
  ],
}
`;

exports[`null code 1`] = `
{
  "colors": {
    "activeTabBackground": "#1E1E1E",
    "activeTabBorder": "#1E1E1E",
    "activeTabForeground": "#ffffff",
    "activeTabTopBorder": undefined,
    "background": "#1E1E1E",
    "colorScheme": "dark",
    "diffInsertedLineBackground": "#9bb95533",
    "diffInsertedTextBackground": "#9ccc2c33",
    "diffRemovedLineBackground": "#ff000033",
    "diffRemovedTextBackground": "#ff000033",
    "editorBackground": "#1E1E1E",
    "editorGroupHeaderBackground": "#252526",
    "foreground": "#D4D4D4",
    "hoverTabBackground": "#2D2D2D",
    "hoverTabForeground": "#ffffff80",
    "iconForeground": "#C5C5C5",
    "inactiveTabBackground": "#2D2D2D",
    "inactiveTabForeground": "#ffffff80",
    "lineNumberForeground": "#858585",
    "listHoverBackground": "#2A2D2E",
    "listHoverForeground": undefined,
    "listSelectionBackground": "#37373D",
    "listSelectionForeground": undefined,
    "selectionBackground": "#264F78",
    "sideBarBackground": "#252526",
    "sideBarBorder": "#252526",
    "sideBarForeground": "#D4D4D4",
    "tabBorder": "#252526",
    "tabsBorder": undefined,
  },
  "lang": "javascript",
  "lines": [
    [
      {
        "content": "",
        "style": {
          "color": "#D4D4D4",
        },
      },
    ],
  ],
}
`;

exports[`null lang 1`] = `
{
  "colors": {
    "activeTabBackground": "#1E1E1E",
    "activeTabBorder": "#1E1E1E",
    "activeTabForeground": "#ffffff",
    "activeTabTopBorder": undefined,
    "background": "#1E1E1E",
    "colorScheme": "dark",
    "diffInsertedLineBackground": "#9bb95533",
    "diffInsertedTextBackground": "#9ccc2c33",
    "diffRemovedLineBackground": "#ff000033",
    "diffRemovedTextBackground": "#ff000033",
    "editorBackground": "#1E1E1E",
    "editorGroupHeaderBackground": "#252526",
    "foreground": "#D4D4D4",
    "hoverTabBackground": "#2D2D2D",
    "hoverTabForeground": "#ffffff80",
    "iconForeground": "#C5C5C5",
    "inactiveTabBackground": "#2D2D2D",
    "inactiveTabForeground": "#ffffff80",
    "lineNumberForeground": "#858585",
    "listHoverBackground": "#2A2D2E",
    "listHoverForeground": undefined,
    "listSelectionBackground": "#37373D",
    "listSelectionForeground": undefined,
    "selectionBackground": "#264F78",
    "sideBarBackground": "#252526",
    "sideBarBorder": "#252526",
    "sideBarForeground": "#D4D4D4",
    "tabBorder": "#252526",
    "tabsBorder": undefined,
  },
  "lang": "javascript",
  "lines": [
    [
      {
        "content": "x",
        "style": {
          "color": "#9CDCFE",
        },
      },
      {
        "content": " = ",
        "style": {
          "color": "#D4D4D4",
        },
      },
      {
        "content": "1",
        "style": {
          "color": "#B5CEA8",
        },
      },
    ],
  ],
}
`;
